engine/src/main/java/com/datatorrent/stram/engine/InputNode.java

```diff
@@ -241,11 +241,14 @@ else if (PROCESSING_MODE == ProcessingMode.EXACTLY_ONCE) {
       if (++applicationWindowCount == APPLICATION_WINDOW_COUNT) {
         applicationWindowCount = 0;
       }
-      if (++checkpointWindowCount == CHECKPOINT_WINDOW_COUNT) {
-        checkpointWindowCount = 0;
-        if (doCheckpoint || PROCESSING_MODE == ProcessingMode.EXACTLY_ONCE) {
-          checkpoint(currentWindowId);
-          lastCheckpointWindowId = currentWindowId;
+
+      if (lastCheckpointWindowId < currentWindowId) {
```

Comment:
can we add a small comment here that why this check is needed?
It helps others to understand